corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js

```diff
@@ -1,5 +1,17 @@
-/* globals hqDefine */
-hqDefine('toggle_ui/js/edit-flag', function () {
+hqDefine('toggle_ui/js/edit-flag', [
+    'jquery',
+    'knockout',
+    'underscore',
+    'hqwebapp/js/initial_page_data',
+    'hqwebapp/js/main',
+    'hqwebapp/js/knockout_bindings.ko',     // save button
+], function (
+    $,
+    ko,
+    _,
+    initialPageData,
+    hqMain
```

Comment:
im probably missing something obvious since I dont have a great grasp of require js but what does it mean here to have more things defined in the input array than this function. there is nothing that matches up to the save button knockout bindings.

Reply:
It's a dependency that doesn't get referenced in the JavaScript. In this case, it's needed because knockout_bindings.ko.js defines a `saveButton` binding that's used by this page. Also happens for jQuery plugins like select2.